language/libra-vm/src/libra_transaction_executor.rs

```diff
@@ -308,23 +309,26 @@ impl LibraVM {
         &self,
         remote_cache: &StateViewCache<'_>,
         writeset_payload: &WriteSetPayload,
+        txn_sender: Option<AccountAddress>,
     ) -> Result<ChangeSet, Result<(VMStatus, TransactionOutput), VMStatus>> {
         let gas_schedule = zero_cost_schedule();
         let mut cost_strategy = CostStrategy::system(&gas_schedule, GasUnits::new(0));
 
         Ok(match writeset_payload {
             WriteSetPayload::Direct(change_set) => change_set.clone(),
-            WriteSetPayload::Script {
-                script,
-                execute_as: signer,
-            } => {
+            WriteSetPayload::Script { script, execute_as } => {
```

Comment:
Maybe we could make `execute_as` an option in another PR? Would that alleviate some of the pain of the all or nothing signers?

Reply:
I don't think execute_as should be an option: if it were `None` it would be basically the same as a normal transaction sent from the Libra Root account.

Reply:
The more I think about it, I think an option will be the right thing. Before this PR, this was the only way to set the signer. Now that we don't always need it, it feels a bit silly

Reply:
I don't think an option here is needed/good. If Libra Root sends an admin script with `execute_as` set to `None` how is that different from Libra Root sending a normal (user) transaction? 

There are cases where an admin script is used, but where there is no sender, in particular for waypoints in the system. So the sender being an option makes sense, but the `execute_as` being an option doesn't no?

Reply:
Ah yes, that makes sense. Maybe cleaning up the `sender`s below will help this crystalize for me